screens/Menu/Settings.js

```diff
@@ -55,14 +140,54 @@ function Settings(props) {
                         <Text style={styles.container2SubText}>Current Status is {switchLabel2}</Text>
                         <View style={styles.toggle}>
                             <Switch
-                                value={switchVal2}
-                                onValueChange={(val) => setSwitchVal2(val)}>
+                                value={switchVal2 == "true" ? true : false}
+                                onValueChange={(val) => {
+                                    AsyncStorage.setItem("switchVal", JSON.stringify(val));
+                                    setSwitchVal2(val.toString());
+                                    signInOrOut(val);
+                                }
+                                }>
                             </Switch>
                         </View>
                     </View>
                 </ScrollView>
             </View>
-        </View >
+            <View style={styles.flatlist}>
+                {(calendarList) &&
+                    <FlatList
+                        data={calendarList}
+                        keyExtractor={(item) => item.id}
+                        renderItem={({ item }) => (
+                            <TouchableOpacity onPress={() => { press(item.id) }}>
+                                <ListItem
+                                    title={item.summary}
+                                    titleStyle={{ color: "white" }}
+                                    subtitle={item.description !== undefined ? item.description : "Please provide a description inside google calendar"}
+                                    rightTitleStyle={{ color: "green" }}
+                                    subtitleStyle={{ color: "grey" }}
+                                    containerStyle={{ backgroundColor: "#2A2E43" }}
+                                    rightIcon={
+                                        <CheckBox
+                                            size={30}
+                                            iconRight
+                                            iconType='material'
+                                            checkedIcon='check'
+                                            uncheckedIcon='add'
+                                            uncheckedColor="grey"
+                                            checkedColor='#3ACCE1'
+                                            onPress={() => {
+                                                press(item.id);
+                                                AsyncStorage.setItem("calendarId", item.id)
```

Comment:
When clicking on the item (not the checkElement), the calendarId is never udpated.

**Solution:** Move the line  `    AsyncStorage.setItem("calendarId", item);` to line `104`. Meaning at the end of the method `press`

Reply:
Its fixed !